src/main/java/org/folio/rest/impl/CodexInvImpl.java

```diff
@@ -71,28 +72,19 @@ private void getUrl(String url, HttpClient client,
         } else {
           fut.handle(Future.failedFuture("Get url " + url + " returned " + res.statusCode()));
         }
+      })
+      .onFailure(r -> {
+        client.close();
+        fut.handle(Future.failedFuture(r.getMessage()));
       });
-    });
-    req.setChunked(true);
-    for (Map.Entry<String, String> e : okapiHeaders.entrySet()) {
-      if (!e.getKey().equalsIgnoreCase(XOkapiHeaders.URL)) {
-        req.putHeader(e.getKey(), e.getValue());
-      }
-    }
-    req.putHeader("Accept", "application/json");
-    req.exceptionHandler(r -> {
-      client.close();
-      fut.handle(Future.failedFuture(r.getMessage()));
-    });
-    req.end();
   }
 
   static IdMaps idMaps = new IdMaps();
 
   private void getMap(Context vertxContext, LHeaders headers, Map<String, String> map,
     String path, String rootElement, Handler<AsyncResult<Void>> fut) {
 
-    HttpClient client = vertxContext.owner().createHttpClient();
+    WebClient client = createWebClient(vertxContext);
```

Comment:
Hmm, this probably shouldn't be creating a new client. I think that is out of scope of this work. 

Hmm, I'm not sure how RAML Module Builder modules ensure only a single client per verticle when they don't implement the verticle.